### PR TITLE
feat: Implemented price set update with prices and aligned pricing API

### PR DESCRIPTION
**Pull request by sradevski**

### CODE REVIEW

File: packages/pricing/integration-tests/__tests__/services/pricing-module/price-set.spec.ts [hunk 1]
```diff
@@ -262,24 +262,67 @@ moduleIntegrationTestRunner({
         })
       })
 
-      describe("update", () => {
+      describe.only("update", () => {
```
Comment (olivermrbl) [inline packages/pricing/integration-tests/__tests__/services/pricing-module/price-set.spec.ts]:
```suggestion
      describe("update", () => {
```

File: packages/pricing/src/services/pricing-module.ts [hunk 2]
```diff
@@ -246,7 +251,104 @@ export default class PricingModuleService<
...
+
+    return isString(idOrSelector) ? priceSets[0] : priceSets
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  protected async update_(
+    data: UpdatePriceSetInput[],
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<PriceSet[]> {
+    // TODO: We are not handling rule types, rules, etc. here, add support after data models are finalized
```
Comment (olivermrbl) [inline packages/pricing/src/services/pricing-module.ts:344]: question: What more is missing about the data models?

File: packages/types/src/pricing/service.ts [hunk 3]
```diff
@@ -602,18 +603,120 @@ export interface IPricingModuleService extends IModuleService {
...
+   *     {
+   *       prices: [{amount: 100, currency_code: "USD"}]
+   *     }
+   *   )
+   *
+   *   // do something with the price set or return it
+   * }
+   */
+  upsert(
+    data: UpsertPriceSetDTO[],
```
Comment (olivermrbl) [inline packages/types/src/pricing/service.ts]:
```suggestion
    data: UpsertPriceSetDTO,
```

File: packages/types/src/pricing/service.ts [hunk 4]
```diff
@@ -602,18 +603,120 @@ export interface IPricingModuleService extends IModuleService {
...
+    sharedContext?: Context
+  ): Promise<PriceSetDTO[]>
```
Comment (olivermrbl) [inline packages/types/src/pricing/service.ts]:
```suggestion
  ): Promise<PriceSetDTO>
```

File: packages/pricing/src/services/pricing-module.ts (hunk 2 above)
Comment (sradevski) [reply packages/pricing/src/services/pricing-module.ts:344]: @olivermrbl rules are not handled at all for price sets currently, among other things. I think it's easier to see the `create` method to see what else is missing, there is a lot going on there that I wouldn't do before Recap

File: packages/types/src/pricing/service.ts (hunk 3 above)
Comment (sradevski) [reply packages/types/src/pricing/service.ts]: Ah nice catch 👍

**Review by olivermrbl — APPROVED** [review]
Mergeable after comments have been addressed 👍

**Pull request merged**